pkg/neotest/compile.go

```diff
@@ -26,6 +26,10 @@ type Contract struct {
 var contracts = make(map[string]*Contract)
 
 // CompileSource compiles a contract from the reader and returns its NEF, manifest and hash.
+// Compiled contract will have "contract.go" used for its file name and coverage
+// data collection can give wrong results for it, so it's recommended to disable
+// coverage ([*Executor.DisableCoverage]) when you're deploying contracts
+// compiled via this function.
```

Comment:
How about enforcing neotest coverage to be disabled when this function is called (and fail test otherwise)? It's harsh for the user, but at the same time it won't lead to unexpected results then.

Reply:
It's not tied to any `Executor` and it can be called without having any `Executor` for some purposes. You can also compile and not deploy.